Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h

```diff
@@ -655,6 +678,7 @@ class MediaPlayerPrivateGStreamer
     RefPtr<PlatformMediaResourceLoader> m_loader;
 
     RefPtr<GStreamerQuirksManager> m_quirksManagerForTesting;
+    HashMap<const GStreamerQuirk*, std::unique_ptr<GStreamerQuirkBase::GStreamerQuirkState>> m_quirkStates;
```

Comment:
I see a couple of problems with this approach:

1. The Broadcom base class will instantiate two structures, no big deal.
2. If we want to specialize the state for the broadcom subclasses, boom.

Anyway, I can live with these problems and solve them later should we need it.

Reply:
No problem. I'm now declaring all the private stuff in GStreamerQuirkBroadcomBase as protected (that's needed to have all the GStreamerQuirkBroadcomBaseState dependencies visible to subclasses) and ensureState() is now virtual. Any subclass can now override ensureState(), with the condition that it creates a (hypothetical) subclass of GStreamerQuirkBroadcomBaseState, sets it on the player private, and returns it cast as GStreamerQuirkBroadcomBaseState.

Anyway, even if the code now solves that problem, it's still a theoretical one, since all the percentage correction family of functions in GStreamerQuirksManager is designed to forward the calls to one (but never more than one) subclass. There should be no clashes between subclasses.